contracts/redeem/src/contract.rs

```diff
@@ -0,0 +1,174 @@
+// use cw2::set_contract_version;
+use crate::error::ContractError;
+use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
+use crate::state::{CfgData, ADMIN, CONFIG};
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{coins, to_json_binary, Addr, BankMsg, Binary, CosmosMsg, Deps, DepsMut, Env, MessageInfo, Order, Response, StdError, StdResult};
+/*
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:redeem";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+*/
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    ADMIN.save(deps.storage, &msg.admin)?;
+    Ok(Response::default())
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::AddConfig { base_asset, stk_asset, exchange_rate } => {
```

Comment:
Given that we are storing the whole struct again during both update and add, one of them is redundant.
`add` will also technically overwrite any older data the way we have written it.